### PR TITLE
ENH: rlz.member_of accepts enum literals now

### PR DESCRIPTION
**Pull request by gerrymanoim**

### CODE REVIEW

**Comment by jreback** [comment]
can you add a release note.

File: ibis/expr/datatypes.py [hunk 1]
```diff
@@ -1714,6 +1715,11 @@ def infer_integer(value: int, allow_overflow: bool = False) -> Integer:
     return int64
 
 
+@infer.register(enum.Enum)
+def infer_enum(value: enum.Enum) -> Enum:
```
Comment (jreback) [inline ibis/expr/datatypes.py:1719]: do we have a direct test of this anywhere?

Comment (gerrymanoim) [reply ibis/expr/datatypes.py:1719]: Err no...it is tested as part of the enum test I added but we should have a direct one. Adding now.

Comment (gerrymanoim) [reply ibis/expr/datatypes.py:1719]: Now tested in https://github.com/ibis-project/ibis/pull/2650/files#diff-752c3dd09545c70e427edd30421b0826bbeb836921c0ca5f6938756e90800ad8R440

**Review by jreback — APPROVED** [review]

**Review by emilyreff7 — APPROVED** [review]
LGTM

**Comment by jreback** [comment]
thanks @gerrymanoim

**Pull request merged**